src/editors/object.js

```diff
@@ -982,10 +988,10 @@ export class ObjectEditor extends AbstractEditor {
 
       this.editors[name] = this.jsoneditor.createEditor(editor, {
         jsoneditor: this.jsoneditor,
-        schema,
+        schema: this.currentRecursions >= this.jsoneditor.MAX_RECURSIONS ? { type: schema.type || null } : schema,
```

Comment:
Let's access to this.jsoneditor.options.maxRecursion (you can destructure it above but let's be consistent with the rest of the codevase